airflow/contrib/operators/jenkins_job_trigger_operator.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import time
+
+from airflow.exceptions import AirflowException
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+from airflow.contrib.hooks.jenkins_hook import JenkinsHook
+import jenkins
+
+try:
+    basestring
+except NameError:
+    basestring = str # For python3 compatibility
+
+class JenkinsJobTriggerOperator(BaseOperator):
+    """
+    Trigger a Jenkins Job and monitor it's execution
+    :param jenkins_connection_id: The jenkins connection to use for this job
+    :type jenkins_connection_id: string
+    :param job_name: The name of the job to trigger
+    :type job_name: string
+    :param parameters: The parameters block to provide to jenkins
+    :type parameters: string
+    :param sleep_time: How long will the operator sleep between each status request for the job (min 1, default 10)
+    :type sleep_time: int
+    :param max_try_before_job_appears: The maximum number of requests to make while waiting for the job to appears on jenkins server (default 10)
+    :type max_try_before_job_appears: int
+    """
+    template_fields = ('parameters',)
+    template_ext = ('.json',)
+    ui_color = '#f9ec86'
+
+    @apply_defaults
+    def __init__(self,
+                 jenkins_connection_id="",
+                 job_name="",
+                 parameters="",
+                 sleep_time=10,
+                 max_try_before_job_appears=10,
+                 *args,
+                 **kwargs):
+        super(JenkinsJobTriggerOperator, self).__init__(*args, **kwargs)
+        self.job_name = job_name
+        self.parameters = parameters
+        if sleep_time < 1:
+            sleep_time = 1
+        self.sleep_time = sleep_time
+        self.jenkins_connection_id = jenkins_connection_id
+        self.max_try_before_job_appears = max_try_before_job_appears
+
+    def get_hook(self):
+        return JenkinsHook(self.jenkins_connection_id)
+
+    def execute(self, context):
+        if not self.jenkins_connection_id:
+            logging.error("Please specify the jenkins connection id to use. You must create a Jenkins connection before being able to use this operator")
```

Comment:
This line is to long, please try to fit in 80 symbols.

Reply:
I tried to cut all the longest lines of the file. 
Is the proper way to do it is to add \ at the end of the line before the line return ?